skmorf/tree/_unsup_criterion.pyx

```diff
@@ -70,12 +69,22 @@ cdef class TwoMeans(Criterion):
             The dataset stored as a buffer for memory efficiency of shape
             (n_samples,).
         """
-        # need to implement
-        pass
-    
+        self.start = 0
+        self.pos = 0
+        self.end = 0
+
+        self.n_samples = 0
+        self.n_node_samples = 0
+        self.weighted_n_node_samples = 0.0
+        self.weighted_n_left = 0.0
+        self.weighted_n_right = 0.0
+
+        cdef SIZE_t k = 0
+        cdef SIZE_t max_n_classes = 0
```

Comment:
You are also missing instantiations for `sum_total, sum_left, sum_right`. See `_unsup_criterion.pxd` file that defines the interface for `UnsupervisedCriterion`.

Reply:
Thank you! I'm still piecing together how to define them b/c those are the counts of classes in supervised setting which we don't have in unsupervised (as they are getting that from `y`). I'll resolve this for now and try again in my next revision :smiley: